### ISSUE TITLE
Check incorp.time = NA and related issues

### ISSUE THREAD

**Issue by sashahafner** [sashahafner/ALFAM2#56]
This problem showed up in v3.2 with ALFAMI work.
Hard to believe function could not handle NAs in incorp variables before this.

**Comment by sashahafner** [comment]
See bcf86c4a8f7eaaf877d5b4ffab09f73182f449e2 and around there

**Comment by sashahafner** [comment]
Related issue fixed in 4fbc47298856a2691f70f28b1d2d398a3ef6d832

prepIncorp relies on sorting but it was not done before calling the hidden function prepIncorp until this commit.

Now with this commit the incorp column will have 0 for the added row and any above it (before incorp time + 1 row) when coming from prepIncorp.
So output from prepIncorp (e.g., par est) cannot be used in a regular (non-flatout) call to alfam2, because the incorp variables are all 0 for the first row and maybe more of the input dataset.
These incorp variables will apparently not be present from the code below--will give 0 for all groups:

```
    # Get number of incorp columns by group for 1) checking that there is not more than 1 applied, 2) skipping incorp when there is none (even when t.incorp may have a value)
    n.incorp.vals <- rowSums(dat[, inc.ex, drop = FALSE])
    n.incorp.vals.grp <- tapply(n.incorp.vals, dat$`__group`, "[", 1)

```

**Comment by sashahafner** [comment]
So never use output from `prepIncorp()` externally as input for `alfam2()` unless `flatout = TRUE` is used (skipping incorp stuff).

**Comment by sashahafner** [comment]
I think flatout issue is addressed with the new approach as of around 792689f96d541c6258071eb46116c6f85eb07378

Here it is from vignette:

```

The issue is preparation of incorporation inputs and determination of parameters.
This is not a simple process, because incorporation occurs at a particular time, which may not correspond to a row in the input data, and may differ among locations.
Large improvements in speed are possible when these steps are skipped, but doing so requires pre-processing of input data.

<<>>=
dat9c <- alfam2(dat9b, app.name = "TAN.app", time.name = "ct", 
                time.incorp = "t.incorp", group = "field", 
                warn = FALSE, value = "incorp")
head(dat9c)
@

Note the new, somewhat strange, columns that have been added.
And note that this pre-processing takes some time.
So separating it only saves time if the actual emission predictions are to be made multiple times.
Examples include parameter estimation or cases where other inputs change, e.g., simulations aimed at the effect of changing weather or slurry properties.

<<>>=
system.time(
  pred9c <- alfam2(dat9c, app.name = 'TAN.app', time.name = 'ct', 
                   time.incorp = "t.incorp", group = 'field', 
                   warn = FALSE, prep.incorp = FALSE, check = FALSE)
)
@

```
